sympy/solvers/solveset.py

```diff
@@ -918,7 +923,13 @@ def solveset(f, symbol=None, domain=S.Complexes):
             result = ConditionSet(symbol, f, domain)
         return result
 
-    return _solveset(f, symbol, domain, _check=True)
+    if decompogen(f, symbol)[0] == f:
+        return _solveset(f, symbol, domain, _check=True)
+    else:
+        try:
+            return solve_decomposition(f, symbol, domain)
+        except:
+            return _solveset(f, symbol, domain, _check=True)
```

Comment:
Why there is `try` block? In which case `solve_decomposition ` is failing. If you really want to handle specific exception then do for only that Exception.